fili-core/src/test/groovy/com/yahoo/bard/webservice/web/endpoints/LookupDimensionFilteringDataServletSpec.groovy

```diff
@@ -0,0 +1,287 @@
+package com.yahoo.bard.webservice.web.endpoints
+
+import com.yahoo.bard.webservice.data.dimension.BardDimensionField
+import com.yahoo.bard.webservice.data.dimension.DimensionDictionary
+import com.yahoo.bard.webservice.util.GroovyTestUtils
+import com.yahoo.bard.webservice.util.JsonSortStrategy
+
+/**
+ * Testing Query Time Lookup with extraction filter on Lookup Dimensions.
+ */
+class LookupDimensionFilteringDataServletSpec extends BaseDataServletComponentSpec{
+
+    @Override
+    def setup() {
+        DimensionDictionary dimensionStore = jtb.configurationLoader.dimensionDictionary
+        dimensionStore.findByApiName("sex").with {
+            addDimensionRow(BardDimensionField.makeDimensionRow(it, "sex1", "sex1Desc"))
+            addDimensionRow(BardDimensionField.makeDimensionRow(it, "sex2", "sex2Desc"))
+            addDimensionRow(BardDimensionField.makeDimensionRow(it, "sex3", "sex3Desc"))
+        }
+        dimensionStore.findByApiName("breed").with {
+            addDimensionRow(BardDimensionField.makeDimensionRow(it, "breed1", "breed1Desc"))
+            addDimensionRow(BardDimensionField.makeDimensionRow(it, "breed2", "breed2Desc"))
+            addDimensionRow(BardDimensionField.makeDimensionRow(it, "breed3", "breed3Desc"))
+        }
+        dimensionStore.findByApiName("species").with {
+            addDimensionRow(BardDimensionField.makeDimensionRow(it, "species1", "species1Desc"))
+            addDimensionRow(BardDimensionField.makeDimensionRow(it, "species2", "species2Desc"))
+            addDimensionRow(BardDimensionField.makeDimensionRow(it, "species3", "species3Desc"))
+        }
+    }
+
+    @Override
+    boolean compareResult(String result, String expectedResult, JsonSortStrategy sortStrategy) {
+        // Using default SORT_MAPS provided by Util
+        GroovyTestUtils.compareJson(result, expectedResult)
+    }
```

Comment:
The default for "test api response" in `BaseDataServletComponentSpec` is `SORT_BOTH`, so we have to override it to use the default sorting mechanism `SORT_MAP` instead by not passing in a sorting mechanism. (or I can add `SORT_MAP`) explicitly.